eZ/Publish/Core/FieldType/Tests/Integration/BinaryBase/BinaryBaseStorage/BinaryBaseStorageTest.php

```diff
@@ -0,0 +1,159 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace Integration\BinaryBase\BinaryBaseStorage;
+
+use eZ\Publish\API\Repository\Values\Content\ContentInfo;
+use eZ\Publish\Core\FieldType\BinaryBase\BinaryBaseStorage;
+use eZ\Publish\Core\FieldType\BinaryBase\BinaryBaseStorage\Gateway;
+use eZ\Publish\Core\FieldType\BinaryFile\BinaryFileStorage\Gateway\DoctrineStorage;
+use eZ\Publish\Core\IO\IOServiceInterface;
+use eZ\Publish\Core\IO\Values\BinaryFile;
+use eZ\Publish\Core\IO\Values\BinaryFileCreateStruct;
+use eZ\Publish\Core\Persistence\Legacy\Tests\TestCase;
+use eZ\Publish\SPI\FieldType\BinaryBase\PathGenerator;
+use eZ\Publish\SPI\IO\MimeTypeDetector;
+use eZ\Publish\SPI\Persistence\Content\Field;
+use eZ\Publish\SPI\Persistence\Content\FieldValue;
+use eZ\Publish\SPI\Persistence\Content\VersionInfo;
+
+class BinaryBaseStorageTest extends TestCase
+{
+    /** @var \eZ\Publish\Core\FieldType\BinaryBase\BinaryBaseStorage\Gateway|\PHPUnit\Framework\MockObject\MockObject */
+    protected $gatewayMock;
+
+    /** @var \eZ\Publish\SPI\FieldType\BinaryBase\PathGenerator|\PHPUnit\Framework\MockObject\MockObject */
+    protected $pathGeneratorMock;
+
+    /** @var \eZ\Publish\Core\IO\IOServiceInterface|\PHPUnit\Framework\MockObject\MockObject */
+    protected $ioServiceMock;
+
+    /** @var \eZ\Publish\Core\FieldType\BinaryBase\BinaryBaseStorage|\PHPUnit\Framework\MockObject\MockObject */
+    protected $storage;
+
+    protected function setUp(): void
+    {
+        $this->gatewayMock = $this->getStorageGateway();
+        $this->pathGeneratorMock = $this->createMock(PathGenerator::class);
+        $this->ioServiceMock = $this->createMock(IOServiceInterface::class);
+        $this->storage = $this->getMockBuilder(BinaryBaseStorage::class)
+            ->setMethods(null)
```

Comment:
setMethods is deprecated. You need to find an alternative (following linked PHPUnit issue in the deprecation notice)